server/routes/offenceRoutes.ts

```diff
@@ -1030,12 +1031,17 @@ export default class OffenceRoutes {
       addOrEditCourtCase,
       addOrEditCourtAppearance,
     } = req.params
+    let offenceConvictionDateForm: OffenceConvictionDateForm
```

Comment:
```suggestion
    const offenceConvictionDateForm = (req.flash('offenceConvictionDateForm')[0] || {}) as OffenceConvictionDateForm
```

need the just submitted form to show values correctly when displaying errors